model/task/db.go

```diff
@@ -799,6 +800,18 @@ func TasksByProjectAndCommitPipeline(projectId, commitHash, taskId, taskStatus s
 		}
 		pipeline = append(pipeline, statusMatch)
 	}
+	if variant != "" {
+		variantMatch := bson.M{
+			"$match": bson.M{BuildVariantKey: variant},
```

Comment:
Rather than adding new $match stages such that there are many matches in a row, can we just save the match stage from the original pipeline as a variable so we can modify it, and only add it to the pipeline after all these if statements?